plutus-core/plutus-ir/src/PlutusIR/Compiler.hs

```diff
@@ -136,14 +135,12 @@ simplifierIteration suffix = do
   costModel <- view ccBuiltinCostModel
   hints <- view (ccOpts . coInlineHints)
   preserveLogging <- view (ccOpts . coPreserveLogging)
-  cocConservative <- view (ccOpts . coCaseOfCaseConservative)
   rules <- view ccRewriteRules
   ic <- view (ccOpts . coInlineConstants)
 
   pure $ P.NamedPass ("simplifier" ++ suffix) $ fold
       [ mwhen (opts ^. coDoSimplifierUnwrapCancel) $ Unwrap.unwrapCancelPass tcconfig
       , mwhen (opts ^. coDoSimplifierCaseReduce) $ CaseReduce.caseReducePass tcconfig
-      , mwhen (opts ^. coDoSimplifierCaseReduce) $ CaseOfCase.caseOfCasePassSC tcconfig binfo cocConservative noProvenance
```

Comment:
This is the only Haskell file changed and this line is the idea of this PR.

Reply:
So only the PIR case-of-case has this problem, the UPLC one doesn't?

Reply:
That's a great question, thank you for asking it! Yes, I believe UPLC case-of-case has one of the sources of exponentiality of PIR case-of-case. I've amended #5966 to reflect that.

Reply:
What about the corresponding CLI flag - Should we remove it too?